### PR TITLE
[AIMIGRAPHX-1003] Add warnings for dim and value not set

### PR DESCRIPTION
**Pull request by eddieliao**
## Motivation
Warn the user when input dimensions or input values are not set, with the goal to improve usability. 

## Technical Details
Batch size is excluded from the warnings as it is common in most models and the default value of 1 typically works.

## Changelog Category

Add a `CHANGELOG.md` entry for any option other than `Not Applicable`
- - [x] Added: New functionality.
- - [ ] Changed: Changes to existing functionality.
- - [ ] Removed: Functionality or support that has been removed. (Compared to a previous release)
- - [ ] Optimized: Component performance that has been optimized or improved.
- - [ ] Resolved Issues: Known issues from a previous version that have been resolved.
- - [ ] Not Applicable: This PR is not to be included in the changelog.

### CODE REVIEW

File: src/driver/main.cpp [hunk 1]
```diff
@@ -593,6 +594,25 @@ struct program_params
...
+            if(not x.empty() and x[0] == '@')
+                load_arg_names.insert(x.substr(1));
+        std::set<std::string> unset;
+        for(const auto& param : param_shapes)
+            if(not contains(param.first, "#output_") and not contains(fill0, param.first) and
+               not contains(fill1, param.first) and not contains(load_arg_names, param.first))
+                unset.insert(param.first);
+        if(unset.empty())
+            return;
+        log::warn() << "Input(s) without explicit values: " << join_strings(std::move(unset), ", ")
```
Comment (pfultz2) [inline src/driver/main.cpp:613]: Not all inputs require explicit values. Most dont require this. I think this warning is just too noisy.

File: src/onnx/onnx_parser.cpp [hunk 2]
```diff
@@ -259,6 +261,31 @@ void onnx_parser::parse_undefined(module* mod, const std::string& name)
...
+    for(const auto& input : graph.input())
+    {
+        if(contains(parser.map_input_dims, input.name()) or
+           contains(parser.map_dyn_input_dims, input.name()))
+            continue;
+        for(const auto& d : input.type().tensor_type().shape().dim())
+        {
+            // Skip batch dims and dims that are already set
+            if(d.has_dim_param() and not contains(parser.dim_params, d.dim_param()) and
+               to_lower(d.dim_param()).find("batch") == std::string::npos)
```
Comment (pfultz2) [inline src/onnx/onnx_parser.cpp]: The batch dimension is not always named batch so I dont think you can rely on that.

File: src/onnx/onnx_parser.cpp [hunk 3]
```diff
@@ -259,6 +261,31 @@ void onnx_parser::parse_undefined(module* mod, const std::string& name)
...
+                unresolved.insert(d.dim_param());
+        }
+    }
+    if(unresolved.empty())
+        return;
+    log::warn() << "Model has unbound symbolic dimension(s): "
```
Comment (pfultz2) [inline src/onnx/onnx_parser.cpp:282]: We shouldn't warn if the user has called `set_default_dim_value` or `set_default_dyn_dim_value`. This is not just a driver warning.

File: src/driver/main.cpp (hunk 1 above)
Comment (eddieliao) [reply src/driver/main.cpp:613]: The idea was to warn when users didn't set values for inputs such as the attention_mask and hopefully reduce the amount of tickets coming to us. Not sure if there's a good list out there for such inputs that require very specific values, but if not I can remove this warning.

File: src/onnx/onnx_parser.cpp (hunk 2 above)
Comment (eddieliao) [reply src/onnx/onnx_parser.cpp]: How often would a batch dimension not contain the word "batch"? The only example I can think of right now would be just using "N", which I can add an edge case for if desired.

File: src/onnx/onnx_parser.cpp (hunk 3 above)
Comment (eddieliao) [reply src/onnx/onnx_parser.cpp:282]: Changed, thanks

File: src/onnx/onnx_parser.cpp (hunk 2 above)
Comment (kahmed10) [reply src/onnx/onnx_parser.cpp]: It could be a variety of names: dim0, b, d0, etc. We shouldn't rely on "batch," but instead if there are unresolved dim params, just log a warning.

Comment (eddieliao) [reply src/onnx/onnx_parser.cpp]: Removed the exception for batch.

File: src/driver/main.cpp (hunk 1 above)
Comment (kahmed10) [reply src/driver/main.cpp:613]: Generally these inputs have `int` datatypes. Would it be possible to log the warning for these cases?

Comment (eddieliao) [reply src/driver/main.cpp:613]: Added check for integer types.

**Review by kahmed10 — APPROVED** [review]

**Review by ahsan-ca — APPROVED** [review]

**Comment by ahsan-ca** [comment]
Nit: Changelog category needs to be selected.
